src/redux/books/books.js

```diff
@@ -1,27 +1,35 @@
-const defaultState = [
-  {
-    id: 1,
-    title: 'The Hunger Games',
-    author: 'Suzanne collins',
-    category: 'Action',
-    progress: 54,
-  },
-  {
-    id: 2,
-    title: 'Dune',
-    author: 'Frank herbert',
-    category: 'Sci-Fi',
-    progress: 64,
-  },
-  {
-    id: 3,
-    title: 'capital In the twenty-First Century',
-    author: 'Suzanne collins',
-    category: 'Economy',
-    progress: 78,
-  },
-];
+const ADD_BOOK = 'bookstore/books/ADD_BOOK';
+const REMOVE_BOOK = 'bookstore/books/REMOVE_BOOK';
 
-export default function books(state = defaultState) {
-  return state;
+const defaultState = [];
+
+export default function books(state = defaultState, action) {
+  switch (action.type) {
+    case ADD_BOOK:
+      return state.concat({
+        id: Date.now(),
+        title: action.payload.title,
+        author: action.payload.author,
+        category: action.payload.category,
+        progress: 0,
```

Comment:
- [ optional ] It will be nice to use the payload directly. Recreating the same object will be redundant especially if you already have the id generated within the component responsible for adding a book.